test/framework/src/main/java/org/elasticsearch/index/mapper/MapperTestCase.java

```diff
@@ -86,8 +86,21 @@
  * Base class for testing {@link Mapper}s.
  */
 public abstract class MapperTestCase extends MapperServiceTestCase {
+
+    public static final IndexVersion DEPRECATED_BOOST_INDEX_VERSION = IndexVersion.V_7_10_0;
+
     protected abstract void minimalMapping(XContentBuilder b) throws IOException;
 
+    /**
+     * Provides a way for subclasses to define minimal mappings for previous index versions
+     * @param b content builder to use for building the minimal mapping
+     * @param indexVersion index version the mapping should be created for
+     * @throws IOException
+     */
+    protected void minimalMapping(XContentBuilder b, IndexVersion indexVersion) throws IOException {
```

Comment:
As discussed, I've added a new `minimalMapping()` method that takes into account index version. This is used in tests that use previous indices (`testDeprecatedBoost`) and can be overridden by subclasses if needed